### PR TITLE
fix(text-input-group): updates to fix when used in dark block

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/5670

### CODE REVIEW

**Review by thatblindgeye — COMMENTED** [review]
Only thing I'm noticing is when dark mode is enable via the switch and then applying `pf-v5-t-light` to `.pf-v5-c-text-input-group`, the icon color becomes black against gray input background. Similar occurrence when applying the `pf-v5-t-light` class to the root when the dark mode switch is off.

Depends how likely a use case that is, but otherwise things look good from a quick look at the preview

**Comment by mcoker** [comment]
@thatblindgeye great catch! Definitely something we should fix. I needed to reset the pf-v5-t-[light/dark] vars in dark theme for the icon vars I added in the previous commit. I also found a block of var declarations that I'm pretty sure we don't need in [sass-utilities/themes/dark/placeholders.scss](https://github.com/patternfly/patternfly/blob/2266ace7039dd918193166282d42ea66f22ab109/src/patternfly/sass-utilities/themes/dark/placeholders.scss), so I removed them, but take a look around and see if you notice anything off. I imagine these are the places we would see issues:

```
cmichael-mac:components cmichael$ grep -ri pf-v5-t- *
AboutModalBox/about-modal-box.scss:  @include pf-v5-t-dark;
Accordion/accordion.scss:  @include pf-v5-t-light;
Alert/alert.scss:  @include pf-v5-t-light; // This component always needs to be light
Banner/banner.scss:  @include pf-v5-t-dark;
Banner/banner.scss:    @include pf-v5-t-light(false);
Banner/banner.scss:    @include pf-v5-t-light(false);
CalendarMonth/calendar-month.scss:  @include pf-v5-t-light;
Chip/chip-group.scss:  @include pf-v5-t-light; // force the container follow the light theme
Chip/chip.scss:  @include pf-v5-t-light;  // This component always needs to be light
ContextSelector/context-selector.scss:  @include pf-v5-t-light;
DataList/data-list.scss:  @include pf-v5-t-light;
FormControl/form-control.scss:  @include pf-v5-t-light("--#{$form-control}--Color");
LogViewer/log-viewer.scss:      @include pf-v5-t-dark;
Login/login.scss:  @include pf-v5-t-dark;
Login/login.scss:  @include pf-v5-t-dark;
Masthead/masthead.scss:  @include pf-v5-t-dark; // force the container to follow the dark theme
Masthead/masthead.scss:    @include pf-v5-t-light;
Masthead/masthead.scss:    @include pf-v5-t-light;
Masthead/masthead.scss:    @include pf-v5-t-dark(--#{$button}--m-plain--Color); // for all plain buttons, especially inside other components (.pf-v5-c-input-group), for dark mode
Menu/menu.scss:  @include pf-v5-t-light;
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
Page/page.scss:  @include pf-v5-t-dark; // force the container to follow the dark theme
Page/page.scss:  @include pf-v5-t-dark;
Page/page.scss:    @include pf-v5-t-light;
Page/page.scss:      @include pf-v5-t-dark;
Page/page.scss:    @include pf-v5-t-dark; // force the container to follow the dark theme
Select/select.scss:  @include pf-v5-t-light; // force the container follow the light theme
Table/table.scss:  @include pf-v5-t-light; // This component always needs to be light
Wizard/wizard.scss:  @include pf-v5-t-dark;
```

**Review by thatblindgeye — APPROVED** [review]
🎉 looks good from what I can tell. One other question is whether `pf-v5-t-light` should override the root theme? If the theme switcher is set to light theme and I apply the dark theme class to the TextInputGroup, it styles that component with its dark theme. If the theme switcher is set to dark, though, applying the light theme class doesn't style the component with its light theme.

This isn't just for TextInputGroup, though, as it looks like the Navigation examples showcasing light theme have the same issue.

File: src/patternfly/base/_variables.scss [hunk 1]
```diff
@@ -232,6 +232,10 @@
   // Icons
   --#{$pf-global}--icon--Color--light: #{$pf-v5-global--icon--Color--light};
   --#{$pf-global}--icon--Color--dark: #{$pf-v5-global--icon--Color--dark};
+  --#{$pf-global}--icon--Color--light--light: #{$pf-v5-global--icon--Color--light--light};
```
Comment (srambach) [inline src/patternfly/base/_variables.scss:235]: I would very much appreciate a comment clarifying which variable is which (e.g. --[light/dark component]--[light/dark theme] or whatever it means :-)

**Comment by srambach** [comment]
Did this break the horizontal nav of the workspace? Or is this PR missing a different fix? 
image

**Comment by thatblindgeye** [comment]
@srambach that's how it currently is on the core and react v5 workspaces, too

**Comment by srambach** [comment]
> @srambach that's how it currently is on the core and react v5 workspaces, too

Ah, my bad - I looked at another preview and thought it was ok there but I was mistaken.

**Review by srambach — APPROVED** [review]
Input in masthead looks good 👍

**Pull request merged**